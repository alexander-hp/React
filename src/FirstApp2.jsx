// TODO si es que la variable no va a cambiar de valor
// TODO o esta asociada a algun hook entonces no debe de ir en
// TODO el functionalcomponent, osea no dentro de la function
const newMessage = [1, 2, 3, 4, 5];

// ?Importacion para las props
import PropTypes from 'prop-types';

// ? Como imprimir un objeto
const message = {
  title: 'Titulo del objeto',
  message: 'mensaje obj',
};

// ? devolvemos un mensaje en una funcion
function getMessage() {
  return 'Mensaje obtenido';
}

const getSuma = (a, b) => {
  return a + b;
};

// ? en el parametro estamos recibiendo las props ya desestructurada
/* 
    ? si es que nosotros asignamos a algo en las props entonces saldra 
    ?  si es que no esta recibiendo ese campo las props, 
    ? pero si ya las tenia antes entonces saldra las que tenia asignadas 
  */
export default function FirstApp2({ title, subtitle, number, boolean }) {
  return (
    // ? esto sustituye al fragment para no anidar codigo a otro div
    // ? el cual podria estropear los estilos que tenemos: <>
    <>
      <h1>Alexander</h1>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      {/* Podemos poner js o variables aqui todo a excepcion de un objeto */}
      {/* ? Pruebas para testear */}
      <h2>{title}</h2>
    </>
  );
}

// *Estamos especificando el tipo de las props
FirstApp2.propTypes = {
  // ? si queremos especificar el tipo y que sea obligatorio:
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  number: PropTypes.number,
};

// *Poniendo el valor que tendran por defecto los props
FirstApp2.defaultProps = {
  title: 'Si no esta asignado',
  subtitle: 'asignado desde funcion',
};
