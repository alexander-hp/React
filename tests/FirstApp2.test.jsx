/** @jest-environment jsdom */
import { render, screen } from '@testing-library/react';
import FirstApp2 from '../src/FirstApp2';

describe('Pruebas en <FirstApp>', () => {
  test('Match con snapshot', () => {
    const title = 'Hola soy el titulo';
    // *Importamos y renderizamos nuestro componente
    const { container } = render(<FirstApp2 title={title} />);
    // * creando y comparando snapshot
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el titulo', () => {
    const title = 'Hola soy el titulo';
    // ? Screen es todo lo que contiene nuestro componente
    render(<FirstApp2 title={title} />);
    // ? Imprimimos lo que tiene screen, que sera nuestro html con el body y demas
    screen.debug();
    // ? checamos que exista el titulo
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('debe de mostrar el titulo en un h1', () => {
    const title = 'Hola soy el titulo';
    // ? Screen es todo lo que contiene nuestro componente
    render(<FirstApp2 title={title} />);
    // ? aqui checamos si hay un h2 que sea igual a title
    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe(title);
  });

  test('debe de mostrar el subtitulo enviado por props', () => {
    const subtitle = 'soy el subtitulo';
    render(<FirstApp2 subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
