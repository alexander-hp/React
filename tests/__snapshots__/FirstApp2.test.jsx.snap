// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp> Match con snapshot 1`] = `
<div>
  <h1>
    Alexander
  </h1>
  <p>
    asignado desde funcion
  </p>
  <p>
    asignado desde funcion
  </p>
  <h2>
    Hola soy el titulo
  </h2>
</div>
`;
